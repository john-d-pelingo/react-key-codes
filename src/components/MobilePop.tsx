import React from 'react'
import { css } from 'emotion'

const MobilePop: React.FC = () => (
  <div aria-label="mobile-pop" className={mobilePopCss}>
    <span className="hello-it-seems">
      Hello! <br />
      It seems like you are browsing this webpage without a keyboard or a mouse.{' '}
      <br />
      Better luck next time!
    </span>
  </div>
)

// -------
// STYLING
// -------

const mobilePopCss = css`
  display: table;
  height: 100%;
  margin: 0 auto;
  text-align: center;
  width: 100%;

  .hello-it-seems {
    display: table-cell;
    font-size: 36px;
    line-height: 40px;
    padding: 0 20px;
    vertical-align: middle;
  }
`

export default MobilePop
