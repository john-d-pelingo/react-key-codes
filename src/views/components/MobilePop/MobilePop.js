import React from 'react';

const MobilePop = () => (
  <div className="mobile-pop">
    <span className="hello-it-seems">
      Hello! <br />It seems like you are browsing this webpage without a keyboard or a mouse. <br />Better luck next time!
    </span>
  </div>
);

export default MobilePop;
