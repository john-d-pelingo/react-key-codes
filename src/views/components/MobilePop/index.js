import MobilePop from './MobilePop';

export default MobilePop;
