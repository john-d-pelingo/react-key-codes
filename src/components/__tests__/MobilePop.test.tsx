import { render } from '@testing-library/react'
import React from 'react'

import { MobilePop } from '../MobilePop'

describe('components - MobilePop', () => {
  it('mounts', () => {
    const { getByLabelText } = render(<MobilePop />)

    const mobilePopElement = getByLabelText('mobile-pop')

    expect(mobilePopElement).toMatchSnapshot()
  })
})
