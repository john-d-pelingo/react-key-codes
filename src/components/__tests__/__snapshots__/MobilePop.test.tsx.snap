// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - MobilePop mounts 1`] = `
.emotion-0 {
  display: table;
  height: 100%;
  margin: 0 auto;
  text-align: center;
  width: 100%;
}

.emotion-0 .hello-it-seems {
  display: table-cell;
  font-size: 36px;
  line-height: 40px;
  padding: 0 20px;
  vertical-align: middle;
}

<div
  aria-label="mobile-pop"
  class="emotion-0"
>
  <span
    class="hello-it-seems"
  >
    Hello! 
    <br />
    It seems like you are browsing this webpage without a keyboard or a mouse.
     
    <br />
    Better luck next time!
  </span>
</div>
`;
